fast_response/FastResponseAnalysis.py

```diff
@@ -992,6 +1020,424 @@ def unblind_TS(self, custom_events=None):
+        return sample
+
+
+    def make_prior_map(self, 
+                        plotting_location=None,
+                        format=True
+                           ):
+        """
+        Generates a zoomed in skymap around the best fit location from the maximum likelihood analysis        
+        """
+        #Format the skymap if told to format (usually formatted at construction so shouldn't change anything)
+        if(format):
+            skymap = self.format_skymap(self.skymap)
+        else:
+            skymap=self.skymap
+        
+        #Plot the prior skymap
+        fig, ax=plt.subplots(1,1,figsize=(10,10))
+        plt.sca(ax)
+        reso=.1
+        sizeZoomInDegs=10
+        nPixZoom=sizeZoomInDegs/reso*60
+        hp.visufunc.gnomview(map=np.log10(skymap),hold=True,title="Prior Skymap",rot=(np.degrees(self.skymap_fit_ra),np.degrees(self.skymap_fit_dec),0),
+            xsize=nPixZoom,ysize=nPixZoom,reso=reso,bgcolor="white",badcolor="white",margins=(.02,.02,.05,0),notext=True,cbar=False)
+            
+        hp.graticule(verbose=True)
+        plotting_utils.plot_labels(self.skymap_fit_dec,self.skymap_fit_ra, reso,nPix=nPixZoom,label_scale=.97)
+        
+        ax=plt.gca()
+        text_objs = ax.findobj(
+                lambda artist: isinstance(artist, Text)
+            )
+        for obj in text_objs:
+            obj.set_fontsize(30)
+        plotting_utils.plot_color_bar(range=[0,np.nanmax(self.skymap)], cmap="viridis", col_label=r"Skymap Prior PDF",
+                    offset=-50,labels=[0, "{0:.1e}".format(np.nanmax(self.skymap)/2),"{0:.2e}".format(np.nanmax(self.skymap))],loc=[0.86, 0.2, 0.03, 0.6])
+        theta, phi =plotting_utils.plot_contours([.9],self.skymap)
+        hp.projplot(theta[0], phi[0], linewidth=2., c='k')
+        for i in range(1, len(theta)):
+            hp.projplot(theta[i], phi[i], linewidth=2., c='k', label=None)
+        name="PriorSkymap"
+        if(not format):
+            name+="Unformatted"
+        plt.savefig(plotting_location+name+".png")
+
+    def _make_posterior_fits(self):
+        #internal function to be called from make_posterior_skymap
+        #saves the skymaps as a fits file
+        extra_header = [('index', self.index),
+                   #('energy_range', ),
+                   #('ns_range', ),
+                   ("Sender","IceCube Collaboration"),
+                   ("TRIGGER_EVENT",self.name),
+                   ("startmjd",self.start),
+                   ("stopmjd",self.stop)
+
+                   ]
+        #startmjd, stopmjd, 
+        hp.write_map(self.analysispath + '/' +'posterior_info.fits',m=self.PosteriorSkymap,
+                    extra_header=extra_header,partial=True    
+                              
+                              )
+   
+        
+    def fluxToAverageNs(self,flux):
+        if(self.inj==None):
+            print("Initialize the injector. Can't convert flux to ns without an injector")
```

Comment:
can you initialize the injector here rather than just throwing an error? this should be a one line function call